aws_helper/config.go

```diff
@@ -188,18 +231,47 @@ func AssumeIamRole(iamRoleOpts options.IAMRoleOptions) (*sts.Credentials, error)
 		sessionDurationSeconds = iamRoleOpts.AssumeRoleDuration
 	}
 
-	input := sts.AssumeRoleInput{
-		RoleArn:         aws.String(iamRoleOpts.RoleARN),
-		RoleSessionName: aws.String(sessionName),
-		DurationSeconds: aws.Int64(sessionDurationSeconds),
-	}
+	if iamRoleOpts.WebIdentityToken != "" {
+		var token string
+		// Check if value is a raw token or a path to a file with a token
+		if _, err := os.Stat(iamRoleOpts.WebIdentityToken); err != nil {
+			token = iamRoleOpts.WebIdentityToken
+		} else {
+			tb, err := os.ReadFile(iamRoleOpts.WebIdentityToken)
+			if err != nil {
+				return nil, err
+			}
+			token = string(tb)
+		}
+		input := sts.AssumeRoleWithWebIdentityInput{
+			RoleArn:          aws.String(iamRoleOpts.RoleARN),
+			RoleSessionName:  aws.String(sessionName),
+			WebIdentityToken: aws.String(token),
+			DurationSeconds:  aws.Int64(sessionDurationSeconds),
+		}
+		req, resp := stsClient.AssumeRoleWithWebIdentityRequest(&input)
+		// InvalidIdentityToken error is a temporary error that can occur
+		// when assuming an Role with a JWT web identity token.
+		// N.B: copied from SDK implementation
+		req.RetryErrorCodes = append(req.RetryErrorCodes, sts.ErrCodeInvalidIdentityTokenException)
+		if err := req.Send(); err != nil {
+			return nil, errors.WithStackTrace(err)
+		}
+		return resp.Credentials, nil
+	} else {
```

Comment:
I think else is not required since `return` is used

Reply:
Thanks for the review. I have removed the unnecessary else clause